### PR TITLE
Date picker: masking date, Fixed code to not allow % key typed in dat…

### PR DESCRIPTION
**Pull request by dmopuri**
…e and time input fields

### CODE REVIEW

**Comment by dmopuri** [comment]
@BenArunski, @mazefate can you please review the date formatting changes

File: src/components/mdpDatePicker/mdpDatePicker.js [hunk 1]
```diff
@@ -461,6 +461,54 @@ module.directive("mdpDatePicker", ["$mdpDatePicker", "$timeout", function($mdpDa
...
                     inputElement.off("reset input blur", onInputElementEvents);
                 });
+
+                if(scope.dateFormat == "MM/DD/YYYY") {
+                    inputElement.on("keypress", formatDateAsTyped);
+                }
+
+
+                function formatDateAsTyped(event) {
+                    var inputDate = this.value;
```
Comment (BenArunski) [inline src/components/mdpDatePicker/mdpDatePicker.js]: I'm not sure what "this" is here.  It would be better to assign `this` to a variable so it is easily referenced and the scope is easier to read.

File: src/components/mdpDatePicker/mdpDatePicker.js [hunk 2]
```diff
@@ -461,6 +461,54 @@ module.directive("mdpDatePicker", ["$mdpDatePicker", "$timeout", function($mdpDa
...
+                    var unicode = event.keyCode ? event.keyCode : event.charCode;
+
+                    //Allow BackSpace, Del, left arrow, right arrow in firefox. IE, chrome, safari supports these keys
+                    if (event.keyCode == 8 || event.keyCode == 46 || (event.keyCode == 37 && !event.charCode) || event.keyCode == 39) {
+                        return true;
+                    }
+
+                    if (isNumber(unicode) && inputDate.length < 10) {
+                            //replaces 2 to 02, 3 to 03.. for month field..
```
Comment (BenArunski) [inline src/components/mdpDatePicker/mdpDatePicker.js]: fix indentation, please.  hard to follow.

File: src/components/mdpDatePicker/mdpDatePicker.js [hunk 3]
```diff
@@ -461,6 +461,54 @@ module.directive("mdpDatePicker", ["$mdpDatePicker", "$timeout", function($mdpDa
...
+                            inputDate = inputDate.replace(/^([2-9])/, "0$1");
+                            //restrict typing 13, 14, 15, 16, 17, 18, 19 for month field
+                            if(inputDate == "1" && (unicode >= 51 && unicode <= 57)) {
+                            return false;
+                            }
+                            //replaces 4 to 04, 5 to 05.. for date field..
+                            inputDate = inputDate.replace(/^([0-9])([0-9])(\/)([4-9])/, "$1$2$30$4");
```
Comment (BenArunski) [inline src/components/mdpDatePicker/mdpDatePicker.js]: Please double-check that this works; I'm not sure about JavaScript's implementation of back-references, but this could be interpreted as "one,two,thirty,four"

File: src/components/mdpDatePicker/mdpDatePicker.js [hunk 4]
```diff
@@ -461,6 +461,54 @@ module.directive("mdpDatePicker", ["$mdpDatePicker", "$timeout", function($mdpDa
...
+                        } else {
+                            return false;
+                        }
+                    }
+
+                function isNumber(unicode) {
+                    var char = String.fromCharCode(unicode);
+
+                    if ((("0123456789").indexOf(char) > -1)) {
+                        // accept input
```
Comment (BenArunski) [inline src/components/mdpDatePicker/mdpDatePicker.js]: comment does not apply to the function.

File: src/components/mdpTimePicker/mdpTimePicker.js [hunk 5]
```diff
@@ -371,8 +371,8 @@ module.directive("mdpTimePicker", ["$mdpTimePicker", "$timeout", function($mdpTi
                 var p = this.value;
                 var unicode = event.keyCode ? event.keyCode : event.charCode;
 
-                //Allow BackSpace, Del, left arrow
-                if(unicode == 8 || unicode == 127 || unicode == 37){
+                //Allow BackSpace, Del, left arrow, right arrow in firefox. IE, chrome, safari supports these keys
+                if(event.keyCode == 8 || event.keyCode == 46 || (event.keyCode == 37 && !event.charCode) || event.keyCode == 39){
```
Comment (BenArunski) [inline src/components/mdpTimePicker/mdpTimePicker.js]: are 39 and 46 the values for "single-quote" and "period"?  And is "Delete"/127 now removed?  What is this change for?

**Comment by BenArunski** [comment]
@dmopuri initial review is complete

File: src/components/mdpDatePicker/mdpDatePicker.js (hunk 3 above)
Comment (dmopuri) [inline src/components/mdpDatePicker/mdpDatePicker.js]: Checked and Its working
